Portal/src/Datahub.Portal/Pages/Tools/Users/UsersPage.razor

```diff
@@ -1,35 +1,65 @@
 @page "/tools/users"
+@using Microsoft.IdentityModel.Tokens
+@using Datahub.Core.Model.Achievements
 
 @inject IDbContextFactory<DatahubProjectDBContext> _dbContextFactory
+@inject IJSRuntime _jsRuntime
 
 <DatahubAuthView AuthLevel="DatahubAuthView.AuthLevels.DatahubAdmin">
-  <MudText Typo="Typo.h1" Class="mb-12">
-      @Localizer["Users"]
-  </MudText>
-  @if (_userWorkspaces == null)
-  {
+    <MudText Typo="Typo.h1" Class="mb-12">
+        @Localizer["Users"]
+    </MudText>
+    @if (_hasLoaded)
+    {   
+        <MudTooltip Text="Copy the emails of all users in the current environment. Excludes gmail, ssc-spc and apption emails.">
+            <MudButton Color="Color.Primary" OnClick="@CopyAllUsersToClipboard">
+                @_allButton
+            </MudButton>
+        </MudTooltip>
+        <MudTooltip Text="Copy the emails of locked users in the tenant.">
+            <MudButton Color="Color.Primary" OnClick="@CopyLockedUsersToClipboard">
+                @_lockedButton
+            </MudButton>
+        </MudTooltip>
+    }
+    else
+    {
+        <MudProgressCircular Indeterminate="true" Color="Color.Primary"/>
+    }
+    @if (_userWorkspaces == null)
+    {
       <MudSkeleton Height="600px" Width="100%"/>
-  }
-  else
-  {
-      <UsersTable UserWorkspaces="_userWorkspaces"/>
-  }
+    }
+    else
+    {
+      <UsersTable UserWorkspaces="_userWorkspaces" @ref="_usersTable"/>
+    }
 </DatahubAuthView>
 
 
 
 @code {
-    
+    private UsersTable _usersTable;
+    private bool _hasLoaded = false;
+    private string _lockedButton = null;
+    private string _allButton = null;
+
     private List<UserWorkspaces> _userWorkspaces;
+    private List<PortalUser> _portalUsers;
+    
     protected override async Task OnAfterRenderAsync(bool firstRender)
     {
-        if (!firstRender) return;
+        if (!firstRender && !_hasLoaded)
+        {
+            _hasLoaded = _usersTable.loadStatus;
+            _lockedButton = "Copy locked users to clipboard";
+            _allButton = "Copy all users to clipboard";
```

Comment:
Don't forget the `Localizer` ! 😄 

Any reason why these are held through variables instead of razor markup?